src/services/xcat.cpp

```diff
@@ -162,24 +197,59 @@ void XCAT::configureTimeService()
 
     m_stateless.postinstall.emplace_back(fmt::format(
         "echo \"server {} iburst\" >> $IMG_ROOTIMGDIR/etc/chrony.conf\n\n",
-        m_cluster->getHeadnode()
+        cluster()
+            ->getHeadnode()
             .getConnection(Network::Profile::Management)
             .getAddress()
             .to_string()));
 }
 
 void XCAT::configureInfiniband()
 {
-    if (const auto& ofed = m_cluster->getOFED())
+    LOG_INFO("[xCAT] Configuring infiniband");
+    if (const auto& ofed = cluster()->getOFED()) {
         switch (ofed->getKind()) {
             case OFED::Kind::Inbox:
                 m_stateless.otherpkgs.emplace_back("@infiniband");
 
                 break;
 
             case OFED::Kind::Mellanox:
-                throw std::logic_error("MLNX OFED is not yet supported");
-
+                {
+                    // Add the rpm to the image
+                    m_stateless.otherpkgs.emplace_back("kmod-mlnx-ofa_kernel");
+                    auto runner = cloyster::Singleton<BaseRunner>::get();
+                    auto arch = cloyster::utils::enumToString(cluster()->getNodes()[0].getOS().getArch());
+
+                    // Cofigure Apache to serve the RPM repository
+                    auto repoFolder = std::string_view("/var/www/html/rpmrepo");
+                    cloyster::installFile(
+                        "/etc/httpd.d/conf.d/rpmrepo.conf",
+                        fmt::format(
+                            R"(Alias "/rpmrepo" "{0}" 
+<Directory "{0}"> 
+    Options +Indexes +FollowSymLinks 
+    AllowOverride None 
+    Require all granted 
+    IndexOptions FancyIndexing VersionSort NameWidth=* HTMLTable Charset=UTF-8 
+</Directory>
+)", repoFolder));
+                    runner->executeCommand("apachectl configtest");
+                    runner->executeCommand("systemctl restart httpd");
+
+                    // Create the RPM repository
+                    runner->executeCommand(
+                        fmt::format("bash -c \"cp -v /usr/share/doca-host-*/Modules/*.{}/*.rpm {}\"", 
+                                    arch, repoFolder));
+                    runner->executeCommand(
+                        fmt::format("createrepo {}", repoFolder));
+
+
+                    // Add the repository to the image
+                    runner->executeCommand(
+                        fmt::format("bash -c \"chdef -t osimage {} --plus otherpkgdir=http://$(hostname)/rpmrepo\"",
+                            m_stateless.osimage));
+                }
```

Comment:
This is the computing node part of the OFED installation,

@viniciusferrao I'm requesting comments here